app/src/main/java/de/westnordost/streetcomplete/location/LocationUtil.java

```diff
@@ -46,15 +34,9 @@ public static boolean hasLocationPermission(Context context)
 
 	public static IntentFilter createLocationAvailabilityIntentFilter()
 	{
-		String action = LocationUtil.isNewLocationApi() ? MODE_CHANGED : PROVIDERS_CHANGED_ACTION;
-		return new IntentFilter(action);
+		return new IntentFilter(MODE_CHANGED);
 	}
 
 	// because LocationManager.MODE_CHANGED is not defined before KitKat
 	private static final String MODE_CHANGED = "android.location.MODE_CHANGED";
```

Comment:
I see more that can be removed here.

Much of `LocationUtil` is a compatibility wrapper

Reply:
What could be removed?

Reply:
I created another PR that removes some of the methods in `LocationUtil`.

Reply:
I think that PR was wrong

Reply:
I added a comment there.